server/src/test/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdaterTests.java

```diff
@@ -151,16 +160,248 @@ public void testDeprecationLogging() {
+        }
+
+        final Settings.Builder existingPersistentSettings = Settings.builder();
+        final Settings.Builder existingTransientSettings = Settings.builder();
+
+        for (final Setting<String> invalidSetting : invalidSettings) {
+            if (randomBoolean()) {
+                existingPersistentSettings.put(invalidSetting.getKey(), "value");
+            } else {
+                existingTransientSettings.put(invalidSetting.getKey(), "value");
+            }
+        }
+
+        for (final Setting<String> unknownSetting : unknownSettings) {
+            if (randomBoolean()) {
+                existingPersistentSettings.put(unknownSetting.getKey(), "value");
+            } else {
+                existingTransientSettings.put(unknownSetting.getKey(), "value");
+            }
+        }
+
+        // register all the known settings (not that we do not register the unknown settings)
+        final Set<Setting<?>> knownSettings =
+                Stream.concat(
+                        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.stream(),
+                        Stream.concat(dynamicSettings.stream(), invalidSettings.stream()))
+                        .collect(Collectors.toSet());
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, knownSettings);
+        for (final Setting<String> dynamicSetting : dynamicSettings) {
+            clusterSettings.addSettingsUpdateConsumer(dynamicSetting, s -> {});
+        }
+        final SettingsUpdater settingsUpdater = new SettingsUpdater(clusterSettings);
+        final MetaData.Builder metaDataBuilder =
+                MetaData.builder()
+                        .persistentSettings(existingPersistentSettings.build())
+                        .transientSettings(existingTransientSettings.build());
+        final ClusterState clusterState = ClusterState.builder(new ClusterName("cluster")).metaData(metaDataBuilder).build();
+
+        // prepare the dynamic settings update
+        final Settings.Builder persistentToApply = Settings.builder();
+        final Settings.Builder transientToApply = Settings.builder();
+        for (final Setting<String> dynamicSetting : dynamicSettings) {
+            if (randomBoolean()) {
+                persistentToApply.put(dynamicSetting.getKey(), "value");
+            } else {
+                transientToApply.put(dynamicSetting.getKey(), "value");
+            }
+        }
+        final ClusterState clusterStateAfterUpdate =
+                settingsUpdater.updateSettings(clusterState, transientToApply.build(), persistentToApply.build(), logger);
+
+        // the invalid settings should be archived
+        for (final Setting<String> invalidSetting : invalidSettings) {
+            if (existingPersistentSettings.keys().contains(invalidSetting.getKey())) {
+                assertThat(
+                        clusterStateAfterUpdate.metaData().persistentSettings().keySet(),
```

Comment:
check that it doesn't exist in the non archived form? alternatively check the total count of settings?

Reply:
I pushed e85ad011637a6c72bb480eb378aad3894d375fbe.